modules/stochastic_tools/test/tests/multiapps/conditional_run/conditional_main.i

```diff
@@ -0,0 +1,68 @@
+[StochasticTools]
+[]
+
+[Distributions]
+  [uniform_0]
+    type = Uniform
+    lower_bound = 0.1
+    upper_bound = 0.3
+  []
+[]
+
+[Samplers]
+  [mc]
+    type = MonteCarlo
+    num_rows = 5
+    distributions = 'uniform_0'
+  []
+  [cart]
+    type = CartesianProduct
+    linear_space_items = '1 1 10'
+  []
+[]
+
+[MultiApps]
+  [runner]
+    type = SamplerFullSolveMultiApp
+    sampler = cart
+    input_files = 'sub.i'
+    mode = batch-reset
+    should_run_reporter = conditional/need_sample
+  []
+[]
+
+[Transfers]
+  [data]
+    type = SamplerReporterTransfer
+    multi_app = runner
+    sampler = cart
+    from_reporter = 'average/value'
+    stochastic_reporter = conditional
+  []
+[]
+
+[Reporters]
+  [conditional]
+    type = ConditionalSampleReporter
+    sampler = cart
+    default_value = 1
+    function = 'val >= t'
+    sampler_vars = 'val'
+    sampler_var_indices = '0'
+    parallel_type = ROOT
```

Comment:
So, the decision as to whether or not to call the subApp is only made on the root proc?

Reply:
No, this is just for outputting. Makes it so the same json file gets created no matter how many processes are used.